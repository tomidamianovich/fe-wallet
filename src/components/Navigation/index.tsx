import React from 'react';
import './styles/index.scss';
import NavItem from '../../components/NavItem';

type Link = { 
  name: string, 
  to: string,
  icon: any,
};

type Props = {
  links: Link[];
};

const Navigation: React.FC<Props> = ({ links }) => 
  <nav className="fe-wallet__content__navigation">
    { 
      links && 
      <ul className="fe-wallet__content__navigation__list">
        { links?.map( (link, index) => <NavItem {...link} key={index} />) }
      </ul> 
    }
    <div className="fe-wallet__content__navigation__logout">
      <NavItem name="Salir" icon="logout" to="logout" />
    </div>
  </nav>
;

export default Navigation;