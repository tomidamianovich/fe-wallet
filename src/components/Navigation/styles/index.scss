@import "../../../containers/App/styles/colours.scss";
@import "../../../containers/App/styles/responsiveness.scss";

.fe-wallet__content__navigation {
  background-color: $secondary-base-color;
  min-width: 256px;
  margin: 0;
  padding: 0;
  height: 100%;

  &__list {
    margin: 0;
    padding: 0;
    list-style: none;
    text-decoration: none;
  }

  &__logout {
    position: absolute;
    bottom: 0;
    padding: 25px 0;
  }

  @media screen and (max-width: $sm) {
    display: none;
  }
}