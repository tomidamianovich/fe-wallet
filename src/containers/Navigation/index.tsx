import React from 'react';
import './styles/index.scss';
import { NavLink } from "react-router-dom";

type Props = {
  links: string[];
};

type LinkItemProps = {
  to: string;
  name: string;
};

const LinkItem = ({to, name}: LinkItemProps) =>  
  <li>
    <NavLink 
      to={to}
      className={({ isActive }) => `fe-wallet__content__navigation__list__item ${isActive && 'fe-wallet__content__navigation__list__item--active'}`}
    >
      {name}
    </NavLink>
  </li>;

const Navigation: React.FC<Props> = ({ links }) => 
  <nav className="fe-wallet__content__navigation">
    <ul className="fe-wallet__content__navigation__list">
      { links.map( (link, index) => <LinkItem to={link} name={link} key={index} />) }
    </ul>
  </nav>
;

export default Navigation;