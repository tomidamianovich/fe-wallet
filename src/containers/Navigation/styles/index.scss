@import "../../../containers/App/styles/colours.scss";
@import "../../../containers/App/styles/responsiveness.scss";

.fe-wallet__content__navigation {
  background-color: $secondary-base-color;
  width: 256px;
  margin: 0;
  padding: 0;
  height: 100%;

  &__list {
    margin: 0;
    padding: 0;
    list-style: none;
    text-decoration: none;
    
    &__item {
      color: black;
      text-decoration: none;
      &--active {
        font-weight: 700;
      }
    }
  }

  @media screen and (max-width: $sm) {
    display: none;
  }
}