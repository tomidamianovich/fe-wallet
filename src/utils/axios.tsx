import { RateType, CurrencyType } from '../utils/type';
const axios = require('axios').default;

type responseType = {
  data: RateType[] | CurrencyType[]
}
  
const requestHandler = async (url:string): Promise<any> => 
  new Promise((resolve, reject) => 
    axios.get(url)
    .then((response: responseType) => resolve(response?.data))
    .catch((err:any) => reject(err))
  );

export {
  requestHandler,
}