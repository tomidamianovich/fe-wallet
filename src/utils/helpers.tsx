const handleFormatValues = (value:number) => value.toFixed(2);

export {
  handleFormatValues
}