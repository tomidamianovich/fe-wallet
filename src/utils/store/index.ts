import currencies from './currencies';
import cryptoCurrencies from './cryptoCurrencies';
import fiatCurrencies from './fiatCurrencies';
import balances from './balances';
import rates from './rates';

export {
  currencies,
  fiatCurrencies,
  cryptoCurrencies,
  rates,
  balances
}