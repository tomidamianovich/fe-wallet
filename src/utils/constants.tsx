import { transaction, transactionTitle, transactionStatus } from '../utils/type';

const ENDPOINTS = {
  CURRENCIES: 'http://localhost:3001/currencies',
  RATES: 'http://localhost:3001/rates',
  BALANCES: 'http://localhost:3001/balance'
}

const PANEL = {
  HEADING: "Mi billetera",
  SUBHEADING: "En tu billetera vas a poder almacenar todas las criptomonedas que compres en ripio.",
}

const TRANSACTIONS:{
  HEADING: string,
  HEADER: string[],
  TYPES: transaction,
  TYPES_TITLES: transactionTitle,
  STATUS: transactionStatus
} = {
  HEADING: "Mis movimientos",
  HEADER: [
    "Fecha",
    "Tipo de Operación",
    "Método",
    "Estado",
    "Comisión",
    "Monto",
  ],
  TYPES: {
    DEPOSIT: "DEPOSIT",
    WITHDRAWAL: "WITHDRAWAL",
    SEND: "SEND",
    RECEIVE: "RECEIVE",
    BUY: "BUY",
    SWAP: "SWAP"
  },
  TYPES_TITLES: {
    DEPOSIT: "Ingreso",
    WITHDRAWAL: "Retiro",
    SEND: "Envio",
    RECEIVE: "Recibo",
    BUY: "Compra",
    SWAP: "Swap"
  },
  STATUS: {
    PENDING: "PENDING",
    FAILURE: "FAILURE",
    SUCCESS: "SUCCESS"
  }
}

const DEFAULT_BALANCE = {
  ticker: "",
  amount: 0,
  account_limit: 0,
  name: "",
  symbol: "",
  decimals: 2,
  type: "",
  actions: [{
    action: "",
    enabled: true
  }],
  url_images: {
    image_png: "",
    image_svg: "",
  },
  color: "",
  categories: [],
  is_favorite: true,
  sell_rate: ""
}

const COMISIONS = {
  BUY: 0.003,
  DEPOSIT: 0,
  WITHDRAWAL: 0.0004,
  SEND: 0,
  RECEIVE: 0,
  SWAP: 0.00004
}

type Link = {
  name: string;
  to: string;
  icon: string;
};

const NAV_LINKS:Link[] = [
  {
    name: 'Panel',
    to: 'panel',
    icon: 'panel'
  },
  {
    name: 'Billetera',
    to: 'wallet',
    icon: 'wallet'
  },
  {
    name: 'Cotizaciones',
    to: 'market-rates',
    icon: 'market-rates'
  },
  {
    name: 'Créditos',
    to: 'credit',
    icon: 'credit'
  },
  {
    name: 'Exchange',
    to: 'exchange',
    icon: 'exchange'
  },
  {
    name: 'Launchpad',
    to: 'launchpad',
    icon: 'launchpad'
  }
];

const WORDINGS = {
  BALANCE: {
    DETAIL: {
      INPUT: {
        LABELS: {
          TRANSACTION: "Seleccione el tipo de operacion a realizar",
          AMOUNT: "Ingrese cantidad",
          ARIA_LABEL : "Monto de la transacción"
        },
        MESSAGES: {
          INSUFFICIENT_AMOUNT: "Dinero insuficiente",
          AMOUNT_TO_DEBIT: "Monto a debitar",
          AMOUNT_NEEDED: "Monto necesario",
          ACTUAL_BALANCE: "Monto actual"
        }
      },
      ACTIONS: {
        CANCEL: {
          LABEL: "Cancelar",
          ARIA_LABEL: "Cancelar transacción",
        },
        CONFIRM: {
          LABEL: "Confirmar",
          ARIA_LABEL: "Confirmar transacción",
        }
      },
      CARD: {
        ACCOUNT: "Cuenta"
      }
    },
    SUMMARY: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar moneda para ver balance"
      }
    }
  },
  CONVERTER: {
    FROM: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar crypto origen",
        NAME: "from"
      },
      INPUT: {
        ARIA_LABEL: "Cripto a convertir",
        NAME: "from_amount"
      }
    },
    TO: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar crypto destino",
        NAME: "to"
      },
      INPUT: {
        ARIA_LABEL: "Cripto convertida",
        NAME: "to_amount"
      }
    },
    BUTTON: {
      ARIA_LABEL: "Convertir",
      LABEL: "Convertir",
    }
  },
  CURRENCIES: {
    TYPES: {
      FIAT: "FIAT",
      CRYPTO: "CRYPTO"
    }
  },
  MAINTENANCE: {
    TITLE: "Pagina en Mantenimiento",
    SUBTITLE: {
      BEGINNING: "Oops! Estamos trabajando en la pagina ",
      ENDING: ". Ingrese nuevamente a esta seccion más tarde."
    }
  },
  NAVIGATION: {
    NAV_ITEM: {
      NAME: "Salir"
    }
  },
  NOT_FOUND: "Pagina no encontrada.",
  ERROR_FOUND: "Oops. Ocurrio un error",
  PAGES: {
    WALLET: "wallet",
    CREDIT: "credit",
    EXCHANGE: "exchange",
    LAUNCHPAD: "launchpad",
    MARKET_RATES: "market-rates",
    PANEL: "panel",
  }
}

export {
  ENDPOINTS,
  PANEL,
  TRANSACTIONS,
  DEFAULT_BALANCE,
  COMISIONS,
  NAV_LINKS,
  WORDINGS
}