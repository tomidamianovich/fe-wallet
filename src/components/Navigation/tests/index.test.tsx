import { ComponentType } from 'react'
import { render } from '@testing-library/react'
import Navigation from '../index'
import { BrowserRouter } from 'react-router-dom'
import { NAV_LINKS } from '../../../utils/constants'

jest.mock('react-i18next', () => ({
  withTranslation: () => (Component: ComponentType) => {
    Component.defaultProps = {
      ...Component.defaultProps,
      t: (translation: string) => translation,
    }
    return Component
  },
}))

test('renders Navigation main without items', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Navigation links={[]} />
    </BrowserRouter>,
  )
  expect(asFragment()).toMatchSnapshot()
})

test('renders Navigation main with items', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Navigation links={NAV_LINKS} />
    </BrowserRouter>,
  )
  expect(asFragment()).toMatchSnapshot()
})
