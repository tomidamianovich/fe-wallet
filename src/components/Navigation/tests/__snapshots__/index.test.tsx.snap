// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navigation main with items 1`] = `
<DocumentFragment>
  <nav
    class="fe-wallet__content__navigation"
  >
    <ul
      class="fe-wallet__content__navigation__list"
    >
      <li
        class="fe-wallet__content__navigation__list__item"
      >
        <a
          class="fe-wallet__content__navigation__list__item__link false"
          href="/fe-wallet"
        >
          <div
            class="fe-wallet__content__navigation__list__item__link__content"
          >
            <svg
              class="icon"
              height="16"
              viewBox="0 0 15 16"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0.471477 7.5952H1.88308V13.7121C1.88308 13.972 2.09374 14.1827 2.35361 14.1827H5.1768C5.43667 14.1827 5.64734 13.972 5.64734 13.7121V9.7126C5.64734 9.06293 6.174 8.53627 6.82367 8.53627C7.47334 8.53627 8 9.06293 8 9.7126V13.7121C8 13.972 8.21066 14.1827 8.47053 14.1827H13.6464C13.9063 14.1827 14.1169 13.972 14.1169 13.7121V7.5952H15.5285C15.963 7.5952 16.1653 7.05666 15.8384 6.77056L8.30985 0.183102C8.13245 0.027874 7.86756 0.027874 7.69015 0.183102L0.161629 6.77056C-0.165345 7.05666 0.0370036 7.5952 0.471477 7.5952ZM8.00007 1.16244L14.2763 6.65413H13.6465C13.3866 6.65413 13.1759 6.8648 13.1759 7.12467V13.2416H8.94113V9.7126C8.94113 8.54319 7.99314 7.5952 6.82374 7.5952C5.65433 7.5952 4.70634 8.54319 4.70634 9.7126V13.2416H2.82421V7.12467C2.82421 6.8648 2.61354 6.65413 2.35368 6.65413H1.72384L8.00007 1.16244Z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              Panel
            </span>
          </div>
        </a>
      </li>
      <li
        class="fe-wallet__content__navigation__list__item"
      >
        <a
          class="fe-wallet__content__navigation__list__item__link false"
          href="/wallet"
        >
          <div
            class="fe-wallet__content__navigation__list__item__link__content"
          >
            <svg
              class="icon"
              height="16"
              viewBox="0 0 15 16"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M13.8666 7.56801V5.43468L3.25504 5.43468C2.85698 5.43597 2.4698 5.32841 2.13331 5.12839V12.7503C2.13531 13.3674 2.63616 13.8667 3.25332 13.8667H13.8666V11.7387H7.71198C7.57335 11.7387 7.44016 11.6847 7.34065 11.5882L5.74065 10.0362C5.52465 9.82666 5.52465 9.48004 5.74065 9.27053L7.34065 7.71853C7.44016 7.622 7.57335 7.56801 7.71198 7.56801H13.8666ZM14.3985 4.36801H12.8566C12.8369 4.33479 12.818 4.301 12.8 4.26668H14.4619C14.7971 4.26668 14.9537 4.5703 14.9312 4.85356C14.9326 4.8693 14.9333 4.88524 14.9333 4.90135V14.4C14.9333 14.6946 14.6945 14.9333 14.4 14.9333H3.25332C2.04841 14.9333 1.07054 13.9586 1.06665 12.752V3.25335C1.06665 3.24682 1.06665 3.24034 1.06692 3.23393C1.06064 2.8415 1.16449 2.44774 1.37845 2.09911C1.79419 1.42172 2.55739 1.02595 3.33124 1.07001H14.4V2.11048C13.7857 2.11048 13.2878 2.59961 13.2878 3.20297C13.2878 3.49195 13.4049 3.76905 13.6133 3.97305C13.8216 4.17706 14.1041 4.2912 14.3983 4.29027L14.3985 4.36801ZM12.4863 3.2029C12.4863 2.81287 12.6007 2.44787 12.8 2.13576L3.33774 2.13508C2.91578 2.11261 2.51519 2.3106 2.29698 2.64946C2.07876 2.98832 2.07876 3.41241 2.29698 3.75127C2.51519 4.09013 2.91578 4.28812 3.36855 4.26483H12.7979C12.5956 3.94623 12.4863 3.57985 12.4863 3.2029ZM14.4615 1.06668C15.0906 1.06668 15.0906 2.13335 14.4615 2.13335L12.8 2.1332C13.1377 1.49442 13.7554 1.06668 14.4615 1.06668ZM6.87798 9.65335L7.92816 10.672H13.8667V8.63468H7.92816L6.87798 9.65335Z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              Billetera
            </span>
          </div>
        </a>
      </li>
      <li
        class="fe-wallet__content__navigation__list__item"
      >
        <a
          class="fe-wallet__content__navigation__list__item__link false"
          href="/market-rates"
        >
          <div
            class="fe-wallet__content__navigation__list__item__link__content"
          >
            <svg
              class="icon"
              height="16"
              viewBox="0 0 15 16"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8 7.91794C5.96079 7.91794 4.30769 6.26484 4.30769 4.22563C4.30769 2.18643 5.96079 0.533325 8 0.533325C10.0392 0.533325 11.6923 2.18643 11.6923 4.22563C11.6923 6.26484 10.0392 7.91794 8 7.91794ZM8 6.68717C9.35947 6.68717 10.4615 5.5851 10.4615 4.22563C10.4615 2.86616 9.35947 1.76409 8 1.76409C6.64053 1.76409 5.53846 2.86616 5.53846 4.22563C5.53846 5.5851 6.64053 6.68717 8 6.68717ZM3.69231 15.3026C1.6531 15.3026 0 13.6495 0 11.6102C0 9.57104 1.6531 7.91794 3.69231 7.91794C5.73151 7.91794 7.38462 9.57104 7.38462 11.6102C7.38462 13.6495 5.73151 15.3026 3.69231 15.3026ZM3.69231 14.0718C5.05178 14.0718 6.15385 12.9697 6.15385 11.6102C6.15385 10.2508 5.05178 9.14871 3.69231 9.14871C2.33284 9.14871 1.23077 10.2508 1.23077 11.6102C1.23077 12.9697 2.33284 14.0718 3.69231 14.0718ZM12.3077 15.3026C10.2685 15.3026 8.61539 13.6495 8.61539 11.6102C8.61539 9.57104 10.2685 7.91794 12.3077 7.91794C14.3469 7.91794 16 9.57104 16 11.6102C16 13.6495 14.3469 15.3026 12.3077 15.3026ZM12.3077 14.0718C13.6672 14.0718 14.7692 12.9697 14.7692 11.6102C14.7692 10.2508 13.6672 9.14871 12.3077 9.14871C10.9482 9.14871 9.84615 10.2508 9.84615 11.6102C9.84615 12.9697 10.9482 14.0718 12.3077 14.0718Z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              Cotizaciones
            </span>
          </div>
        </a>
      </li>
      <li
        class="fe-wallet__content__navigation__list__item"
      >
        <a
          class="fe-wallet__content__navigation__list__item__link false"
          href="/credit"
        >
          <div
            class="fe-wallet__content__navigation__list__item__link__content"
          >
            <svg
              class="icon"
              height="16"
              viewBox="0 0 15 16"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.2136 4.20637C14.4889 4.21062 14.7472 4.06868 14.8907 3.83161C15.0406 3.58405 15.0406 3.27372 14.8907 3.02615C14.7408 2.77858 14.4658 2.63475 14.177 2.65283C13.9022 2.67002 13.6656 2.46124 13.6484 2.1865C13.6312 1.91175 13.8399 1.67508 14.1147 1.65789C14.7738 1.61663 15.4014 1.94487 15.7434 2.50983C16.0855 3.07479 16.0855 3.78298 15.7434 4.34794C15.4934 4.76088 15.0909 5.04735 14.635 5.15551C14.7365 5.4773 14.8036 5.81355 14.8319 6.16087C14.9178 7.14835 14.6942 8.13649 14.1945 8.98903C14.1794 9.05589 14.1494 9.1217 14.1029 9.18249L11.5537 12.5099C11.2978 12.8391 10.9042 13.0316 10.4873 13.0316C10.0703 13.0316 9.67675 12.8391 9.41904 12.5076L8.61167 11.4563H7.87264L7.06767 12.5099C6.81176 12.8391 6.41819 13.0316 6.00126 13.0316C5.60389 13.0316 5.22774 12.8567 4.97177 12.5553C4.94335 12.5311 4.90651 12.4991 4.86204 12.4595C4.74897 12.359 4.62359 12.2426 4.48894 12.1113C4.10534 11.7374 3.72203 11.3176 3.36335 10.8602C2.73658 10.0608 2.25351 9.23506 1.96851 8.39584L1.34986 8.39579C0.996836 8.4022 0.655842 8.26753 0.402458 8.02164C0.149074 7.77575 0.00423104 7.43896 0 7.07998V6.30146C0.00423718 5.94849 0.148497 5.61165 0.401073 5.36505C0.653649 5.11845 0.993849 4.9823 1.34081 4.98652H1.85839C2.15103 4.07283 2.64514 3.18595 3.28531 2.33861C3.65487 1.84946 4.05003 1.40446 4.44558 1.0111C4.65923 0.798642 4.83228 0.640694 4.94678 0.54302C5.20275 0.241591 5.57893 0.0666656 5.97634 0.0666656C6.39327 0.0666656 6.78684 0.259202 7.04532 0.591693L7.84772 1.64196H9.61996C11.5651 1.64196 13.3159 2.6683 14.2136 4.20637ZM13.2574 8.61075C13.7068 7.90827 13.9112 7.07684 13.8387 6.24607L13.8384 6.24306C13.6741 4.21889 11.8058 2.63885 9.61996 2.63885H7.60126C7.44594 2.63885 7.29949 2.56644 7.20519 2.44301L6.25573 1.20023C6.18868 1.114 6.08557 1.06355 5.97634 1.06355C5.86711 1.06355 5.76399 1.114 5.69695 1.20023L5.62175 1.27786C5.60147 1.29469 5.55906 1.33128 5.49771 1.38653C5.3929 1.48089 5.27545 1.59175 5.14853 1.71797C4.7845 2.07997 4.42008 2.49035 4.08071 2.93955C3.42684 3.80501 2.94811 4.70572 2.71576 5.60912C2.6591 5.82941 2.46049 5.9834 2.23303 5.9834L1.33481 5.98337C1.24624 5.9823 1.16087 6.01647 1.09749 6.07835C1.03411 6.14023 0.997906 6.22475 0.996852 6.30739V7.07406C0.997893 7.16166 1.03383 7.24522 1.0967 7.30623C1.15957 7.36724 1.24418 7.40066 1.34081 7.39899H2.3377C2.5633 7.39899 2.76078 7.55051 2.81916 7.76843C3.03863 8.58755 3.50552 9.42583 4.14786 10.2451C4.47726 10.6652 4.83118 11.0528 5.18481 11.3975C5.30819 11.5178 5.42241 11.6238 5.52436 11.7145C5.58413 11.7676 5.62553 11.803 5.64545 11.8194L5.72187 11.898C5.78892 11.9843 5.89203 12.0347 6.00126 12.0347C6.11049 12.0347 6.21361 11.9843 6.27808 11.9014L7.23011 10.6553C7.32441 10.5318 7.47086 10.4594 7.62619 10.4594H8.85734C9.01223 10.4594 9.15832 10.5314 9.25266 10.6543L10.2079 11.898C10.2749 11.9843 10.378 12.0347 10.4873 12.0347C10.5965 12.0347 10.6996 11.9843 10.7645 11.9008L13.2071 8.71259C13.2196 8.67797 13.2362 8.64384 13.2574 8.61075ZM10.0038 3.28184C10.279 3.28184 10.5022 3.505 10.5022 3.78028C10.5022 4.05557 10.279 4.27873 10.0038 4.27873H7.60126C7.32598 4.27873 7.10282 4.05557 7.10282 3.78028C7.10282 3.505 7.32598 3.28184 7.60126 3.28184H10.0038ZM4.94954 6.00833C4.4624 6.00833 4.06232 5.6185 4.06232 5.13107C4.06232 4.64363 4.4624 4.25381 4.94954 4.25381C5.43669 4.25381 5.83677 4.64363 5.83677 5.13107C5.83677 5.6185 5.43669 6.00833 4.94954 6.00833Z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              Cr√©ditos
            </span>
          </div>
        </a>
      </li>
      <li
        class="fe-wallet__content__navigation__list__item"
      >
        <a
          class="fe-wallet__content__navigation__list__item__link false"
          href="/exchange"
        >
          <div
            class="fe-wallet__content__navigation__list__item__link__content"
          >
            <svg
              class="icon"
              height="16"
              viewBox="0 0 15 16"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.32318 3.01615H15.3835C15.7233 3.01615 15.9988 3.29163 15.9988 3.63144V6.70789C15.9988 7.04771 15.7233 7.32318 15.3835 7.32318H0.616523C0.0683586 7.32318 -0.206163 6.66043 0.181448 6.27282L6.27282 0.181448C6.66043 -0.206163 7.32318 0.0683586 7.32318 0.616523V3.01615ZM6.70789 4.24673C6.36808 4.24673 6.0926 3.97126 6.0926 3.63144V2.10196L2.10196 6.0926H14.7682V4.24673H6.70789ZM0.616523 12.9223C0.276708 12.9223 0.00123351 12.6468 0.00123351 12.307V9.23058C0.00123351 8.89077 0.276708 8.61529 0.616523 8.61529H15.3835C15.9316 8.61529 16.2062 9.27804 15.8186 9.66566L9.72718 15.757C9.33957 16.1446 8.67682 15.8701 8.67682 15.3219V12.9223H0.616523ZM1.23181 11.6917H9.29211C9.63192 11.6917 9.9074 11.9672 9.9074 12.307V13.8365L13.898 9.84587H1.23181V11.6917Z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              Exchange
            </span>
          </div>
        </a>
      </li>
      <li
        class="fe-wallet__content__navigation__list__item"
      >
        <a
          class="fe-wallet__content__navigation__list__item__link false"
          href="/launchpad"
        >
          <div
            class="fe-wallet__content__navigation__list__item__link__content"
          >
            <svg
              class="icon"
              height="16"
              viewBox="0 0 15 16"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4.9227 12.3067H3.07669C1.37748 12.3067 0 10.9293 0 9.23006V3.07669C0 1.37748 1.37748 0 3.07669 0H12.9221C14.6213 0 15.9988 1.37748 15.9988 3.07669V9.23006C15.9988 10.9293 14.6213 12.3067 12.9221 12.3067H9.48494L5.97314 15.8185C5.5855 16.2062 4.9227 15.9316 4.9227 15.3834V12.3067ZM8.79495 11.2563C8.91035 11.1409 9.06686 11.0761 9.23006 11.0761H12.9221C13.9416 11.0761 14.7681 10.2496 14.7681 9.23006V3.07669C14.7681 2.05716 13.9416 1.23067 12.9221 1.23067H3.07669C2.05716 1.23067 1.23067 2.05716 1.23067 3.07669V9.23006C1.23067 10.2496 2.05716 11.0761 3.07669 11.0761H5.53803C5.87788 11.0761 6.15337 11.3516 6.15337 11.6914V13.8979L8.79495 11.2563ZM8.64086 4.09115C8.34227 3.92887 8.23177 3.55526 8.39404 3.25666C8.55632 2.95807 8.92993 2.84757 9.22853 3.00985C9.44752 3.12886 9.73148 3.36499 10.0024 3.74639C10.4302 4.34857 10.6823 5.13884 10.6823 6.12261C10.6823 7.10637 10.4302 7.89665 10.0024 8.49882C9.73148 8.88022 9.44752 9.11635 9.22853 9.23536C8.92993 9.39764 8.55632 9.28714 8.39404 8.98855C8.23177 8.68995 8.34227 8.31634 8.64086 8.15406C8.70493 8.11925 8.84554 8.00232 8.99917 7.78608C9.27901 7.39218 9.45158 6.85101 9.45158 6.12261C9.45158 5.3942 9.27901 4.85303 8.99917 4.45913C8.84554 4.24289 8.70493 4.12597 8.64086 4.09115ZM6.86716 5.53803C6.52732 5.53803 6.25182 5.26254 6.25182 4.9227C6.25182 4.58286 6.52732 4.30736 6.86716 4.30736C7.207 4.30736 7.4825 4.58286 7.4825 4.9227C7.4825 5.26254 7.207 5.53803 6.86716 5.53803ZM6.86716 7.99938C6.52732 7.99938 6.25182 7.72389 6.25182 7.38405C6.25182 7.04421 6.52732 6.76871 6.86716 6.76871C7.207 6.76871 7.4825 7.04421 7.4825 7.38405C7.4825 7.72389 7.207 7.99938 6.86716 7.99938Z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              Launchpad
            </span>
          </div>
        </a>
      </li>
    </ul>
    <div
      class="fe-wallet__content__navigation__logout"
    >
      <li
        class="fe-wallet__content__navigation__list__item"
      >
        <a
          class="fe-wallet__content__navigation__list__item__link false"
          href="/logout"
        >
          <div
            class="fe-wallet__content__navigation__list__item__link__content"
          >
            <svg
              class="icon"
              height="16"
              viewBox="0 0 15 16"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M2.39378 14.1768H7.9752C8.31814 14.1768 8.59534 14.454 8.59534 14.7969C8.59534 15.1398 8.31814 15.417 7.9752 15.417H2.39378C1.36805 15.417 0.533325 14.5823 0.533325 13.5566V2.39381C0.533325 1.36808 1.36805 0.533325 2.39378 0.533325H7.9752C8.31814 0.533325 8.59534 0.810584 8.59534 1.15353C8.59534 1.49647 8.31814 1.77367 7.9752 1.77367H2.39378C2.05148 1.77367 1.77364 2.05148 1.77364 2.39381V13.5566C1.77364 13.8989 2.05145 14.1768 2.39378 14.1768ZM11.5113 3.81271L15.2819 7.53365C15.4003 7.65087 15.4667 7.80901 15.4667 7.97514C15.4667 8.14133 15.3997 8.30012 15.2819 8.4167L11.5113 12.1376C11.391 12.2567 11.2335 12.3162 11.076 12.3162C10.916 12.3162 10.7559 12.2542 10.6344 12.1314C10.3944 11.8877 10.3963 11.4952 10.6406 11.2546L13.3352 8.59534H6.11472C5.77177 8.59534 5.49458 8.31811 5.49458 7.97517C5.49458 7.63223 5.77177 7.35503 6.11472 7.35503H13.3352L10.6406 4.69582C10.3963 4.45519 10.3938 4.06264 10.6344 3.81892C10.875 3.5752 11.2682 3.57208 11.5113 3.81271Z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              navigation.nav_item.name
            </span>
          </div>
        </a>
      </li>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`renders Navigation main without items 1`] = `
<DocumentFragment>
  <nav
    class="fe-wallet__content__navigation"
  >
    <ul
      class="fe-wallet__content__navigation__list"
    />
    <div
      class="fe-wallet__content__navigation__logout"
    >
      <li
        class="fe-wallet__content__navigation__list__item"
      >
        <a
          class="fe-wallet__content__navigation__list__item__link false"
          href="/logout"
        >
          <div
            class="fe-wallet__content__navigation__list__item__link__content"
          >
            <svg
              class="icon"
              height="16"
              viewBox="0 0 15 16"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M2.39378 14.1768H7.9752C8.31814 14.1768 8.59534 14.454 8.59534 14.7969C8.59534 15.1398 8.31814 15.417 7.9752 15.417H2.39378C1.36805 15.417 0.533325 14.5823 0.533325 13.5566V2.39381C0.533325 1.36808 1.36805 0.533325 2.39378 0.533325H7.9752C8.31814 0.533325 8.59534 0.810584 8.59534 1.15353C8.59534 1.49647 8.31814 1.77367 7.9752 1.77367H2.39378C2.05148 1.77367 1.77364 2.05148 1.77364 2.39381V13.5566C1.77364 13.8989 2.05145 14.1768 2.39378 14.1768ZM11.5113 3.81271L15.2819 7.53365C15.4003 7.65087 15.4667 7.80901 15.4667 7.97514C15.4667 8.14133 15.3997 8.30012 15.2819 8.4167L11.5113 12.1376C11.391 12.2567 11.2335 12.3162 11.076 12.3162C10.916 12.3162 10.7559 12.2542 10.6344 12.1314C10.3944 11.8877 10.3963 11.4952 10.6406 11.2546L13.3352 8.59534H6.11472C5.77177 8.59534 5.49458 8.31811 5.49458 7.97517C5.49458 7.63223 5.77177 7.35503 6.11472 7.35503H13.3352L10.6406 4.69582C10.3963 4.45519 10.3938 4.06264 10.6344 3.81892C10.875 3.5752 11.2682 3.57208 11.5113 3.81271Z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              navigation.nav_item.name
            </span>
          </div>
        </a>
      </li>
    </div>
  </nav>
</DocumentFragment>
`;
